File: backend/app/schemas/voc.py
from datetime import datetime

from pydantic import BaseModel, Field


class VisitVocCreate(BaseModel):
    company_name: str = Field(min_length=1, max_length=160)
    visit_date: str = Field(min_length=1, max_length=20)
    visitor_name: str = Field(min_length=1, max_length=100)
    contact_name: str | None = Field(default=None, max_length=100)
    channel: str = Field(default="방문", max_length=30)
    sentiment: str = Field(default="보통", max_length=30)
    product_area: str | None = Field(default=None, max_length=80)
    voc_text: str = Field(min_length=1)
    next_action: str | None = None


class VisitVocRead(VisitVocCreate):
    id: str
    created_at: datetime
    status: str


class MeetingMinutesCreate(BaseModel):
    title: str = Field(min_length=1, max_length=160)
    company_name: str | None = Field(default=None, max_length=160)
    meeting_date: str | None = Field(default=None, max_length=20)
    participants: str | None = Field(default=None, max_length=255)
    transcript_text: str = Field(min_length=1)


class MeetingMinutesRead(BaseModel):
    title: str
    company_name: str | None
    meeting_date: str | None
    participants: list[str]
    source_file_name: str | None
    summary: str
    key_topics: list[str]
    decisions: list[str]
    action_items: list[str]
    risks: list[str]
    original_transcript: str

File: frontend/src/api/voc.ts
import { apiClient } from './client';

export type VisitVocPayload = {
  company_name: string;
  visit_date: string;
  visitor_name: string;
  contact_name?: string;
  channel: string;
  sentiment: string;
  product_area?: string;
  voc_text: string;
  next_action?: string;
};

export type VisitVoc = VisitVocPayload & {
  id: string;
  created_at: string;
  status: string;
};

export type MeetingMinutes = {
  title: string;
  company_name?: string | null;
  meeting_date?: string | null;
  participants: string[];
  source_file_name?: string | null;
  summary: string;
  key_topics: string[];
  decisions: string[];
  action_items: string[];
  risks: string[];
  original_transcript: string;
};

export async function createVisitVoc(payload: VisitVocPayload) {
  const response = await apiClient.post<VisitVoc>('/voc/entries', payload);
  return response.data;
}

export async function createMinutesFromUpload(payload: {
  title: string;
  company_name?: string;
  meeting_date?: string;
  participants?: string;
  transcript_text?: string;
  recording_file?: File;
}) {
  const formData = new FormData();
  formData.append('title', payload.title);
  if (payload.company_name) formData.append('company_name', payload.company_name);
  if (payload.meeting_date) formData.append('meeting_date', payload.meeting_date);
  if (payload.participants) formData.append('participants', payload.participants);
  if (payload.transcript_text) formData.append('transcript_text', payload.transcript_text);
  if (payload.recording_file) formData.append('recording_file', payload.recording_file);

  const response = await apiClient.post<MeetingMinutes>('/voc/minutes/upload', formData, {
    headers: { 'Content-Type': 'multipart/form-data' },
  });
  return response.data;
}
